Cannot use paths with spaces in options param

# Introduction
I am using this action in my CI/CD pipeline, where, specifically for Windows, I am having trouble setting a flag in the `options` parameter.

## Description
The problematic parameter is `CUDA_TOOLKIT_ROOT_DIR`. This param is set previously by the action `Jimver/cuda-toolkit`, however, that's not very relevant, as the real cause of the issue is that the default installation path contains spaces (`C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8`, for example).

The error received is:
```
CMake Error: Parse error in command line argument: Files/NVIDIA
 Should be: VAR:type=value
```

I have tried, for no avail, to add quotes around the path (simple and/or double), to scape the spaces with a backwards slash, to replace all backwards slashes with forward ones, and all possible combinations of all the options mentioned.

## Cause of the issue
Diving a bit inside this action's code, I found [this line](https://github.com/threeal/cmake-action/blob/5754e2e49b65c655ea007d6da160d525383c4100/dist/action.mjs#L104), which splits the `options` parameter by spaces.
That makes sense if you have multiple options set, but it does not take into account that the value of any of those options could contain spaces, which is the case for some paths.

## Suggestion
My suggestion to fix this issue would be to detect quotes around the right side of the `=`, so that the spaces inside such value are copied to the final cmake command instead of being used as a separator.
For example, for this options:
```yml
...
options: |
            CMAKE_BUILD_TYPE=${{ env.BUILD_TYPE }}
            CUDA_TOOLKIT_ROOT_DIR="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8"
```
The following parameters would be generated:
```
-DCMAKE_BUILD_TYPE=Release -DCUDA_TOOLKIT_ROOT_DIR="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8"
```
(keeping the quotes to avoid errors down the road).

I now get another error related to CUDA not being found.
```
-- Building for: Visual Studio 17 2022
-- The C compiler identification is MSVC 19.42.34433.0
-- The CXX compiler identification is MSVC 19.42.34433.0
CMake Error at C:/Program Files/CMake/share/cmake-3.31/Modules/CMakeDetermineCompilerId.cmake:614 (message):
  No CUDA toolset found.
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.31/Modules/CMakeDetermineCompilerId.cmake:8 (CMAKE_DETERMINE_COMPILER_ID_BUILD)
  C:/Program Files/CMake/share/cmake-3.31/Modules/CMakeDetermineCompilerId.cmake:53 (__determine_compiler_id_test)
  C:/Program Files/CMake/share/cmake-3.31/Modules/CMakeDetermineCUDACompiler.cmake:131 (CMAKE_DETERMINE_COMPILER_ID)
  CMakeLists.txt:26 (project)
-- Configuring incomplete, errors occurred!
Error: Command failed: cmake D:\a\my-test-project\my-test-project -B D:\a\my-test-project\my-test-project/build -DCMAKE_C_COMPILER=cl -DCMAKE_CXX_COMPILER=cl -DCMAKE_BUILD_TYPE=Release -DCUDA_TOOLKIT_ROOT_DIR=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8 -DCMAKE_CUDA_COMPILER=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8\bin\nvcc.exe
```
However, looks more cmake related and not necessarily to this action which looks like it parsed the arguments correctly.

Thanks for the fix!

--- a/package.json
+++ b/package.json
@@ -9,15 +9,18 @@
     "test": "jest"
   },
   "dependencies": {
-    "gha-utils": "^0.4.1"
+    "gha-utils": "^0.4.1",
+    "shell-quote": "^1.8.1"
   },
   "devDependencies": {
     "@eslint/js": "^9.15.0",
     "@jest/globals": "^29.7.0",
+    "@rollup/plugin-commonjs": "^28.0.1",
     "@rollup/plugin-node-resolve": "^15.3.0",
     "@rollup/plugin-typescript": "^12.1.1",
     "@types/jest": "^29.5.14",
     "@types/node": "^22.9.0",
+    "@types/shell-quote": "^1",
     "eslint": "^9.14.0",
     "jest": "^29.7.0",
     "prettier": "^3.3.3",

--- a/rollup.config.js
+++ b/rollup.config.js
@@ -1,3 +1,4 @@
+import commonjs from "@rollup/plugin-commonjs";
 import { nodeResolve } from "@rollup/plugin-node-resolve";
 import typescript from "@rollup/plugin-typescript";
 
@@ -7,5 +8,5 @@ export default {
     dir: "dist",
     entryFileNames: "[name].mjs",
   },
-  plugins: [nodeResolve(), typescript()],
+  plugins: [commonjs(), nodeResolve(), typescript()],
 };

--- a/src/context.test.ts
+++ b/src/context.test.ts
@@ -116,12 +116,12 @@ describe("get action context", () => {
     },
     {
       name: "with additional arguments specified",
-      inputs: { args: "-Wdev -Wdeprecated\n--fresh" },
+      inputs: { args: `-Wdev -Wdeprecated\n--fresh --foo "bar baz"` },
       expectedContext: {
         configure: {
           generator: "",
           options: [],
-          args: ["-Wdev", "-Wdeprecated", "--fresh"],
+          args: ["-Wdev", "-Wdeprecated", "--fresh", "--foo", "bar baz"],
         },
       },
     },
@@ -151,7 +151,7 @@ describe("get action context", () => {
         "c-flags": "-Werror -Wall\n-Wextra",
         "cxx-flags": "-Werror -Wall\n-Wextra  -Wpedantic",
         options: `BUILD_TESTING=ON BUILD_EXAMPLES=ON\nBUILD_DOCS=ON FOO="BAR BAZ"`,
-        args: "-Wdev -Wdeprecated\n--fresh",
+        args: `-Wdev -Wdeprecated\n--fresh --foo "bar baz"`,
         "run-build": "true",
         "build-args": "--target foo\n--parallel  8",
       },
@@ -170,7 +170,7 @@ describe("get action context", () => {
             "BUILD_DOCS=ON",
             "FOO=BAR BAZ",
           ],
-          args: ["-Wdev", "-Wdeprecated", "--fresh"],
+          args: ["-Wdev", "-Wdeprecated", "--fresh", "--foo", "bar baz"],
         },
         build: {
           enabled: true,

--- a/src/context.test.ts
+++ b/src/context.test.ts
@@ -132,11 +132,11 @@ describe("get action context", () => {
     },
     {
       name: "with additional build arguments specified",
-      inputs: { "build-args": "--target foo\n--parallel  8" },
+      inputs: { "build-args": `--target foo\n--parallel  8 --foo "bar baz"` },
       expectedContext: {
         build: {
           enabled: false,
-          args: ["--target", "foo", "--parallel", "8"],
+          args: ["--target", "foo", "--parallel", "8", "--foo", "bar baz"],
         },
       },
     },
@@ -153,7 +153,7 @@ describe("get action context", () => {
         options: `BUILD_TESTING=ON BUILD_EXAMPLES=ON\nBUILD_DOCS=ON FOO="BAR BAZ"`,
         args: `-Wdev -Wdeprecated\n--fresh --foo "bar baz"`,
         "run-build": "true",
-        "build-args": "--target foo\n--parallel  8",
+        "build-args": `--target foo\n--parallel  8 --foo "bar baz"`,
       },
       expectedContext: {
         sourceDir: "project",
@@ -174,7 +174,7 @@ describe("get action context", () => {
         },
         build: {
           enabled: true,
-          args: ["--target", "foo", "--parallel", "8"],
+          args: ["--target", "foo", "--parallel", "8", "--foo", "bar baz"],
         },
       },
     },

--- a/src/context.test.ts
+++ b/src/context.test.ts
@@ -99,12 +99,17 @@ describe("get action context", () => {
     {
       name: "with additional options specified",
       inputs: {
-        options: "BUILD_TESTING=ON BUILD_EXAMPLES=ON\nBUILD_DOCS=ON",
+        options: `BUILD_TESTING=ON BUILD_EXAMPLES=ON\nBUILD_DOCS=ON FOO="BAR BAZ"`,
       },
       expectedContext: {
         configure: {
           generator: "",
-          options: ["BUILD_TESTING=ON", "BUILD_EXAMPLES=ON", "BUILD_DOCS=ON"],
+          options: [
+            "BUILD_TESTING=ON",
+            "BUILD_EXAMPLES=ON",
+            "BUILD_DOCS=ON",
+            "FOO=BAR BAZ",
+          ],
           args: [],
         },
       },
@@ -145,7 +150,7 @@ describe("get action context", () => {
         "cxx-compiler": "clang++",
         "c-flags": "-Werror -Wall\n-Wextra",
         "cxx-flags": "-Werror -Wall\n-Wextra  -Wpedantic",
-        options: "BUILD_TESTING=ON BUILD_EXAMPLES=ON\nBUILD_DOCS=ON",
+        options: `BUILD_TESTING=ON BUILD_EXAMPLES=ON\nBUILD_DOCS=ON FOO="BAR BAZ"`,
         args: "-Wdev -Wdeprecated\n--fresh",
         "run-build": "true",
         "build-args": "--target foo\n--parallel  8",
@@ -163,6 +168,7 @@ describe("get action context", () => {
             "BUILD_TESTING=ON",
             "BUILD_EXAMPLES=ON",
             "BUILD_DOCS=ON",
+            "FOO=BAR BAZ",
           ],
           args: ["-Wdev", "-Wdeprecated", "--fresh"],
         },

--- a/src/context.ts
+++ b/src/context.ts
@@ -49,9 +49,7 @@ export function getContext(): Context {
     configure: {
       generator: getInput("generator"),
       options,
-      args: getInput("args")
-        .split(/\s+/)
-        .filter((arg) => arg != ""),
+      args: parse(getInput("args")).map((arg) => arg.toString()),
     },
     build: {
       enabled: getInput("run-build") == "true",

--- a/src/context.ts
+++ b/src/context.ts
@@ -53,9 +53,7 @@ export function getContext(): Context {
     },
     build: {
       enabled: getInput("run-build") == "true",
-      args: getInput("build-args")
-        .split(/\s+/)
-        .filter((arg) => arg != ""),
+      args: parse(getInput("build-args")).map((arg) => arg.toString()),
     },
   };
 }

--- a/src/context.ts
+++ b/src/context.ts
@@ -1,5 +1,6 @@
 import { getInput } from "gha-utils";
 import path from "node:path";
+import { parse } from "shell-quote";
 
 export interface Context {
   sourceDir: string;
@@ -39,10 +40,7 @@ export function getContext(): Context {
 
   input = getInput("options");
   if (input) {
-    const opts = input.split(/\s+/).filter((arg) => arg != "");
-    for (const opt of opts) {
-      options.push(opt);
-    }
+    options.push(...parse(input).map((opt) => opt.toString()));
   }
 
   return {

